src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -55,27 +59,48 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
       fileIdTracker: FileIdTracker): Option[Relation] = {
     logicalRelation.relation match {
       case HadoopFsRelation(
-          location: PartitioningAwareFileIndex,
-          _,
-          dataSchema,
-          _,
-          fileFormat,
-          options) if isSupportedFileFormat(fileFormat) =>
+      location: PartitioningAwareFileIndex,
+      _,
+      dataSchema,
+      _,
+      fileFormat,
+      options) if isSupportedFileFormat(fileFormat) =>
         val files = location.allFiles
         // Note that source files are currently fingerprinted when the optimized plan is
         // fingerprinted by LogicalPlanFingerprint.
         val sourceDataProperties =
-          Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
+        Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
         val fileFormatName = fileFormat.asInstanceOf[DataSourceRegister].shortName
+
         // "path" key in options can incur multiple data read unexpectedly.
         val opts = options - "path"
+
+        val rootPaths = opts.get(GLOBBING_PATTERN_KEY) match {
+          case Some(pattern) =>
+            // Validate if globbing pattern matches actual source paths.
+            // This logic is picked from the globbing logic at:
+            // https://github.com/apache/spark/blob/v2.4.4/sql/core/src/main/scala/org/apache/
+            // spark/sql/execution/datasources/DataSource.scala#L540
+            val globPaths = pattern.split(",").map(_.trim).map { path =>
+              val hdfsPath = new Path(path)
+              val fs = hdfsPath.getFileSystem(new Configuration)
+              val qualified = hdfsPath.makeQualified(fs.getUri, fs.getWorkingDirectory)
+              qualified.toString -> SparkHadoopUtil.get.globPathIfNecessary(fs, qualified)
+            }.toMap
+
+            val globPathValues = globPaths.values.flatten.toSet
+            if (!location.rootPaths.forall(globPathValues.contains)) {
+              throw HyperspaceException("Some glob patterns do not match with available root " +
+                s"paths of the source data. Please check if option $GLOBBING_PATTERN_KEY matches " +
+                "with the provided paths.")
```

Comment:
Should we log globbing pattern and root paths for easier debugging?

Reply:
thanks @imback82 ,added